docs/en/operations/server-configuration-parameters/settings.md

```diff
@@ -2191,6 +2191,48 @@ Default: false.
 </clickhouse>
 ```
 
+## replicated_fetches_log {#replicated-fetches-log}
+
+Logging replicated fetch events. You can use the events in this table to analyze the data replicating issues (network bandwidth, fetch speed slow or failing, etc...).
```

Comment:
But the same information about fetched parts is already present in `system.part_log` (entries with type `DOWNLOAD_PART`), why have a separate table then?

Reply:
@kssenii perhaps the most important information is which replica the part was downloaded from and how long it took. We use this table to understand the network traffic during replication in order to optimise the server geo-topology.

Reply:
Consider adding more columns to part_log

Reply:
@canhld94 , hello, are you planning to add more columns to `system.part_log` instead of adding a separate table? If not, we shall close this PR as it seems abandoned now.

Reply:
@kssenii sorry, I've been working on other issues. Actually before I didn't prefer adding new column to system.part_log because the amount of code changes will be relatively large (need to pass part_log_writer to DataExchange methods).
I will check more to see if there's a more simple way to add necessary information to part_log table and try to give final comment in few days [THANKS].

Reply:
Hi @tavplubix @kssenii, I've looking at this PR today, and I think it's better to have a separate table. Not only because of adding more column to `system.part_log` is a lot more complicated, but also in some cases records in `system.part_log` don't have corresponding 1-1 mapping to `system.replicated_fetches`. Example:

- In `executeReplaceRange`, we may fetch many parts but in part_log it's recorded as new parts.
- In `executeClonePartFromShard` we may fetch part but not write part log, and then later when the part is attached it will be recorded as new part.
- In `executeFetch`, if we execute fetch to detached folder (e.g. `FETCH PARTITION`) then we won't write part log for fetched part.
- In `fetchExistsPart` we only write part log if there's exception.

Another pros from my experience is it's easy to use same queries (or with very little modification) for both `system.replicated_fetches` and `system.replicated_fetches_log` for analysis.

Hope this can convince you that this system table is helpful!

Reply:
> In executeReplaceRange, we may fetch many parts but in part_log it's recorded as new parts.

Let's fix that and record them properly.

> In executeClonePartFromShard we may fetch part but not write part log, and then later when the part is attached it will be recorded as new part.

Let's fix that and record it as fetched part

> In executeFetch, if we execute fetch to detached folder (e.g. FETCH PARTITION) then we won't write part log for fetched part.

Let's add another entry type for `part_log` (smth like `DOWNLOAD_PART_TO_DETACHED`)

> In fetchExistsPart we only write part log if there's exception.

Let's fix that (we have `exception` column in part_log for logging unsuccessful attempts)

Reply:
Hello @canhld94! Are you planning to resolve comments from above or should we close this PR?